source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,417 @@
+following:
+
+.. code-block:: json
+
+   { _id: 98765, orders: [] }
+
+The documents in the ``inventory`` collection in this example contain the
+following:
+
+.. code-block:: json
+
+   [
+    { name: "sunblock", sku: 5432, qty: 85 },
+    { name: "beach towel", sku: 7865, qty: 41 }
+   ]
+
+The code examples also perform operations on the ``orders`` collection, but
+do not require any prior sample documents.
+
+The code examples use the ``cart`` and ``payment`` variables to represent
+a sample list of items purchased and the order payment details as follows:
+
+.. code-block:: javascript
+
+   const cart = [
+     { name: 'sunblock', sku: 5432, qty: 1, price: 5.19 },
+     { name: 'beach towel', sku: 7865, qty: 2, price: 15.99 }
+   ];
+   const payment = { customer: 98765, total: 37.17 };
+
+.. important::
+
+   The examples in the following sections require that you create the
+   collections outside of the transaction or that you are using MongoDB 4.4 or
+   later. For more information on creating collections inside a transaction,
+   see the
+   :manual:`Create Collections and Indexes in a Transaction </core/transactions/#std-label-transactions-create-collections-indexes>`
+   server guide.
+
+.. _nodejs-transaction-core-api-example:
+
+Core API Implementation
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The code example in this section demonstrates how you can use the Core API to
+run the multi-document payment transaction in a session. This function
+shows how you can perform the following:
+
+1. Start a session
+#. Start a transaction, specifying transaction options
+#. Perform data operations in the same session
+#. Commit a transaction, or cancel it if the driver encounters an error
+#. End a session
+
+.. literalinclude:: /code-snippets/transactions/txn-core.js
+   :language: javascript
+   :linenos:
+   :emphasize-lines: 2-6,8,10,19,32,41,49,51,55-59,63,65
+   :start-after: start placeOrder
+   :end-before: end placeOrder
+
+Note that you must pass the session object to each CRUD operation that
+you want to run on that session.
+
+The code and comments in the ``catch`` block demonstrate how you can identify
+the server transaction errors and where you can place your logic to handle
+them. Make sure to include the ``MongoError`` type from the driver in your
+code as shown in the following sample import statement:
+
+.. code-block:: javascript
+
+   const { MongoError, MongoClient } = require('mongodb');
+
+See the :ref:`Payment Transaction Result <nodejs-transaction-example-payment-result>`
+section to see what your collections should contain after you run the
+transaction successfully.
+
+.. _nodejs-transaction-callback-api-example:
+
+Callback API Implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The code examples in this section show how you can use the Callback API to
+run the multi-document payment transaction in a session.
+
+The following code example shows a function you can use to start the
```

Comment:
[suggestion] very minor, you can take or leave this - the following example doesn't show a function in the same sense that the other examples do, it's more like the contents of a function that the user could write.

I think this line could be reduced to something like "The following code example shows how you can start the session and pass the callback..."